parlai/tasks/wizard_of_wikipedia/agents.py

```diff
@@ -382,19 +388,62 @@ def get(self, episode_idx, entry_idx=0):
         text = dialog_entry_1['text']
         labels = [dialog_entry_2['text']]
 
+        if self.opt['add_topic'] and entry_idx == 0:
+            text = d.get('chosen_topic', '') + '\n' + text
+
         action = {
             'id': 'WizardBasicDialog',
             'text': text,
             'labels': labels,
             'episode_done': episode_done,
         }
+        if 'label_candidates' in d:
+            action['label_candidates'] = d['label_candidates']
 
         if self.speaker_label == 'wizard':
             action['chosen_topic'] = d.get('chosen_topic', '')
 
         return action
 
 
+class BasicWizardDialogTeacher(BasicdialogTeacher):
+    @staticmethod
+    def add_cmdline_args(argparser):
+        agent = argparser.add_argument_group('Basic Dialog Arguments')
+        agent.add_argument(
+            '--speaker-label',
+            type=str,
+            default='wizard',
+            choices=['both', 'wizard', 'apprentice'],
+            help='Which speaker labels to train on',
+        )
+        agent.add_argument(
+            '--add-topic',
+            type='bool',
+            default=True,
+            help='prepend chosen topic to first turn',
+        )
+
+
+class BasicApprenticeDialogTeacher(BasicdialogTeacher):
+    @staticmethod
+    def add_cmdline_args(argparser):
+        agent = argparser.add_argument_group('Basic Dialog Arguments')
+        agent.add_argument(
```

Comment:
hmm, instead of changing the argument defaults (which is a bit vague/hidden), can we just override the init and hard code `self.speaker_label = 'apprentice'`?

Reply:
ok, done

Reply:
ping